packages/bff/src/model/api/converters/catalogClientApiConverter.ts

```diff
@@ -0,0 +1,214 @@
+      id: eservice.producerId,
+      name: producerTenant.name,
+    },
+    isMine: isRequesterEqProducer,
+    hasCertifiedAttributes,
+  };
+
+  return {
+    ...partialEnhancedEservice,
+    ...(activeDescriptor
+      ? {
+          activeDescriptor: {
+            id: activeDescriptor.id,
+            version: activeDescriptor.version,
+            audience: activeDescriptor.audience,
+            state: activeDescriptor.state,
+          },
+        }
+      : {}),
+    ...(agreement
+      ? {
+          agreement: {
+            id: agreement.id,
+            state: agreement.state,
+            canBeUpgraded: isUpgradable(agreement),
+          },
+        }
+      : {}),
+  };
+}
+
+export function toBffCatalogApiDescriptorAttribute(
+  attributes: attributeRegistryApi.Attribute[],
+  descriptorAttributes: catalogApi.Attribute[]
+): bffApi.DescriptorAttribute[] {
+  return descriptorAttributes.map((attribute) => {
+    const foundAttribute = attributes.find((att) => att.id === attribute.id);
+    if (!foundAttribute) {
+      throw attributeNotExists(unsafeBrandId(attribute.id));
+    }
+
+    return {
+      id: attribute.id,
+      name: foundAttribute.name,
+      description: foundAttribute.description,
+      explicitAttributeVerification: attribute.explicitAttributeVerification,
+    };
+  });
+}
+
+export function toBffCatalogApiDescriptorDoc(
+  document: catalogApi.EServiceDoc
+): bffApi.EServiceDoc {
+  return {
+    id: document.id,
+    name: document.name,
+    contentType: document.contentType,
+    prettyName: document.prettyName,
+  };
+}
+
+export function toBffCatalogApiEserviceRiskAnalysis(
+  riskAnalysis: catalogApi.EServiceRiskAnalysis
+): bffApi.EServiceRiskAnalysis {
+  const answers: { [key: string]: string[] } =
+    riskAnalysis.riskAnalysisForm.singleAnswers
+      .concat(
+        riskAnalysis.riskAnalysisForm.multiAnswers.flatMap((multiAnswer) =>
+          multiAnswer.values.map((answerValue) => ({
+            id: multiAnswer.id,
+            value: answerValue,
+            key: multiAnswer.key,
+          }))
+        )
+      )
+      .reduce((answers: { [key: string]: string[] }, answer) => {
+        const key = `${answer.key}`;
+        if (answers[key] && answer.value) {
+          answers[key] = [...answers[key], answer.value];
+        } else {
+          answers[key] = [];
+        }
+
+        return answers;
+      }, {});
+
+  const riskAnalysisForm: bffApi.RiskAnalysisForm = {
+    riskAnalysisId: riskAnalysis.id,
+    version: riskAnalysis.riskAnalysisForm.version,
+    answers,
+  };
+
+  return {
+    id: riskAnalysis.id,
+    name: riskAnalysis.name,
+    createdAt: riskAnalysis.createdAt,
+    riskAnalysisForm,
+  };
+}
+
+export function toBffCatalogApiProducerDescriptorEService(
+  eservice: catalogApi.EService,
+  producer: tenantApi.Tenant
+): bffApi.ProducerDescriptorEService {
+  const producerMail = producer.mails.find(
+    (m) => m.kind === tenantMailKind.ContactEmail
```

Comment:
I know the values are the same and that TS probably helps with this, but here you are comparing enums of different types: `tenantApi.Tenant` and our internal `TenantMailKind`. To be safer I would refactor as follows:
```suggestion
    (m) => m.kind === tenantApi.MailKind.Values.CONTACT_EMAIL
```